src/pkg/lint/findings.go

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package lint contains functions for verifying zarf yaml files are valid
+package lint
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/defenseunicorns/pkg/helpers/v2"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/fatih/color"
+)
+
+// PackageFinding is a struct that contains a finding about something wrong with a package
+type PackageFinding struct {
+	// YqPath is the path to the key where the error originated from, this is sometimes empty in the case of a general error
+	YqPath      string
+	Description string
+	// Item is the value of a key that is causing an error, for example a bad image name
+	Item string
+	// PackageNameOverride shows the name of the package that the error originated from
+	// If it is not set the base package will be used when displaying the error
+	PackageNameOverride string
+	// PackagePathOverride shows the path to the package that the error originated from
+	// If it is not set the base package will be used when displaying the error
+	PackagePathOverride string
+	Severity            Severity
+}
+
+// Severity is the type of finding
+type Severity int
+
+// different severities of package errors
+const (
+	SevErr Severity = iota + 1
+	SevWarn
+)
+
+func (f PackageFinding) itemizedDescription() string {
+	if f.Item == "" {
+		return f.Description
+	}
+	return fmt.Sprintf("%s - %s", f.Description, f.Item)
+}
+
+func colorWrapSev(s Severity) string {
```

Comment:
We could change this rater but UI stuff should not be here.

Reply:
Hmm pretty much the entirety of this file is UI stuff. Do you feel we should move this entire file minus the data types to message like `printCredentialsTable`?

Reply:
Keeping this here until we have a better solution as moving to message would require keeping the lint types in the types package which we want to avoid